src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -111,7 +114,18 @@ protected void runTool(final JavaSparkContext ctx) {
         //NOTE: this doesn't honor enabled/disabled commandline filters
         final ReadFilter bqsrReadFilter = ReadFilter.fromList(BaseRecalibrator.getBQSRSpecificReadFilterList(), getHeaderForReads());
 
-        final JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+        JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+
+        if (joinStrategy.equals(JoinStrategy.OVERLAPS_PARTITIONER)) {
+            // the overlaps partitioner requires that reads are coordinate-sorted
+            final SAMFileHeader readsHeader = getHeaderForReads().clone();
+            readsHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+            ReadCoordinateComparator comparator = new ReadCoordinateComparator(readsHeader);
+            markedFilteredReadsForBQSR = markedFilteredReadsForBQSR
+                    .mapToPair(read -> new Tuple2<>(read, null))
+                    .sortByKey(comparator, true, numReducers)
+                    .keys();
+        }
 
         VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
         JavaRDD<GATKVariant> bqsrKnownVariants = variantsSparkSource.getParallelVariants(baseRecalibrationKnownVariants, getIntervals());
```

Comment:
Yes, it will. To avoid it we'll need to create our own RDD, e.g. SortedRDD, which remembers the "sortedness". I think this is a bigger refactoring that deserves its own issue.